x-pack/test/observability_ai_assistant_api_integration/tests/knowledge_base/knowledge_base_user_instructions.spec.ts

```diff
@@ -88,6 +88,7 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         });
 
         await Promise.all(promises);
+        await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
Ideally we can avoid - can you reproduce the flakiness? usually things like this are related to Elasticsearch not having refreshed the shard yet. Can you figure out if we are waiting for a refresh in the endpoint? (Look for `refresh: 'wait_for'`.

Reply:
Hi @dgieselaar 
Thank you for the review.
I wasn't able to reproduce the flakiness.. I tried reproducing locally and via the flaky test runner, but all tests passed.

In the code for these tests, I don't see `refresh: 'wait_for'`

This is the output I get after locally running the tests:
<img width="993" alt="image" src="https://github.com/user-attachments/assets/00599c8f-8ce8-4b07-b39f-6fc9987657bc">

Reply:
@viduni94 , Dario was referring to checking whether the actual endpoint we're using querys the knowledge base with `refresh: wait_for`.  We can see that it is, here https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts#L662.  

[Looking at the docs ](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html), it says: 

> Never start multiple refresh=wait_for requests in a row. Instead batch them into a single bulk request with refresh=wait_for and Elasticsearch will start them all in parallel and return only when they have all finished.

Perhaps this could be causing some flakiness.  Bulk indexing would be more reliable. I was thinking we can instead use `POST /internal/observability_ai_assistant/kb/entries/import` when we're wanting to index multiple documents for testing, but it looks like that this not actually bulk indexing but creating multiple requests with concurrency control.

Perhaps for now we can use the `retry` service but think about improving the API to bulk requests.  CC @dgieselaar

Reply:
Thank you @neptunian 
I've updated the PR with the suggested changes.

Reply:
Sorry for the misunderstanding but when I said to perhaps try a retry, I meant to retry fetching them if one of the entries was missing, not to try adding the entries again.  The idea behind that was because they are not yet available due to the multiple requests potentially.